lazer/sdk/js/src/util.ts

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable unicorn/prefer-global-this */
+// we disable this line, because the "self" variable is a
+// construct that is unique to browser workers: https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/self
+// the "globalThis" rule is disabled for a similar reason, but this one
+// is specific to node, but also exists in various browser contexts: 
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis
+
+import type { Data } from 'isomorphic-ws';
+
+/**
+ * Utility function for detecting whether the current JS execution
+ * context is that of a background WebWorker or a ServiceWorker (not the main UI thread)
+ */
+export function envIsWorker() {
+  const possiblyInAWorker = typeof WorkerGlobalScope !== 'undefined' && typeof self !== 'undefined';
+  return possiblyInAWorker && self instanceof WorkerGlobalScope;
+}
+
+/**
+ * Detects if the code is running in a regular DOM or Web Worker context.
+ * @returns true if running in a DOM or Web Worker context, false if running in Node.js
+ */
+export function envIsBrowser() {
+  // disabling this rule for this line, as the rule is reporting a false positive (it's a bug)
+  // for the typeof check for the existence of the querySelector function
+
```

Comment:
looks like eslint decided this rule wasn't actually in use and removed the disable....